docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,475 @@
+All keys and values are treated as case-insensitive.
+
+For example:
+
+```ini
+[Model] # Special section that describes model itself
+Type = NeuralNetwork  # Model Type : only NeuralNetwork is supported as of now
+Optimizer = adam  # Optimizer : sgd (stochastic gradient decent),
+          #             adam (Adaptive Moment Estimation)
+#### optimizer related properties
+Learning_rate = 0.0001  # Learning Rate for the optimizer
+Decay_rate = 0.96 # The decay rate for decaying the learning rate
+Decay_steps = 1000       # decay step for the exponentially decayed learning rate
+beta1 = 0.9     # beta 1 for adam
+beta2 = 0.9999  # beta 2 for adam
+epsilon = 1e-7  # epsilon for adam
+
+#### Model compile related properties
+batch_size = 9
+loss = cross      # Cost(loss) function : mse (mean squared error)
+                        #                       cross (cross entropy)
+
+####  Model run related properties
+Epochs = 20     # Epochs
+save_path = "model.bin"   # model path to save and read parameters
+
+[DataSet] # Special section that describes dataset.
+BufferSize=9
+TrainData="trainingSet.dat"
+ValidData="validationSet.dat"
+LabelData="label.dat"
+
+# Layer Sections, each section name refers to name of the layer
+[inputlayer]
+Type = input
+Input_Shape = 1:1:62720 # Input Layer Dimension in channel:height:width
+Normalization = true
+
+[outputlayer]
+Type = fully_connected
+Unit = 2    # Output Layer Dimension ( Weight Width )
+bias_initializer = zeros
+weight_initializer = xavier_uniform
+Activation = sigmoid  # activation : sigmoid, softmax
+weight_regularizer = l2norm
+weight_regularizer_constant = 0.005
+```
+
+The following restrictions must be adhered to:
+
+ - Model file must have a `[Model]` section.
+ - Model file must have at least one layer.
+ - Valid keys must have valid properties. The invalid keys in each section is ignored.
+
+> [!NOTE]
+> All paths are relative to the current working directory unless stating an absolute path.
+> It is recommended to set `save_path` and `Dataset` from the code over describing inside the model file.
+
+Following example constructs model from INI file:
+
+```c
+char *res_path = app_get_resource_path();
+char model_path[1024];
+ml_train_model_h model;
+
+snprintf(model_path, sizeof(model_path), "%s/model.ini", res_path);
+free(res_path);
+
+status = ml_train_model_construct_with_conf(model_path, &model);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+```
+
+### Construct a model on code
+
+An empty model can be constructed with `ml_train_model_construct()`.
+
+## Configure the model
+
+After constructing a model, the model can be configured.
+
+> [!NOTE]
+> Example code written here reproduces the model description from [Create Model from INI Formatted File](#create-a-model-from-ini-formatted-file) except that followings are different:
+> - Relative path is changed to dynamic app resource and data path.
+> - Model related properties (that can only be set at compile or run phase)
```

Comment:
Suggestion:
- Model related properties that can only be set at compile or run phase.

Reply:
Done